third-party/realdds/src/dds-topic-reader-thread.cpp

```diff
@@ -67,16 +75,22 @@ void dds_topic_reader_thread::run( qos const & rqos )
                     eprosima::fastdds::dds::SampleLostStatus status;
                     _reader->get_sample_lost_status( status );
                     on_sample_lost( _reader, status );
+                    if( stopped->get_trigger_value() )
```

Comment:
I think the `if`s are mutual exclusive, we don't need to repeat this 3 times just add it once after the `if`s.

Reply:
They are not mutually exclusive, though. You can have multiple conditions triggered. You think not? If so, we should have an `else` in there...

Reply:
I assumed that if a sample is lost than data is not available, but I guess it can be about two different samples in one callback.

Reply:
It's just a bit-mask, so it can be that we got data, so we mark data-available. However, we also notice that the data is not the expected sequence number - so we mark sample-list. And then we can also notice that we our subscription was matched. I.e., three different events all at once.